src/com/cft/contactmerge/NameMatchLogic.java

```diff
@@ -0,0 +1,129 @@
+                    {"Al", "Allen"},
+                    {"Al", "Alan"},
+                    {"Bob", "Robert"}
+            };
+
+    /***********************************************************************
+     ********************* Names Compare Logic Methods *********************
+     ***********************************************************************/
+    protected static boolean isNamesExactMatch(ArrayList<String> nameOne, ArrayList<String> nameTwo){
+        return nameOne.equals(nameTwo);
+    }
+
+    protected static boolean doNamesContainMultipleParts(ArrayList<String> nameOne, ArrayList<String> nameTwo){
+        return nameOne.size() > 1 || nameTwo.size() > 1;
+    }
+
+    protected static boolean doNamesContainMultipleParts(ArrayList<String> nameOne, ArrayList<String> nameTwo, ArrayList<String> nameThree, ArrayList<String> nameFour){
+        return nameOne.size() > 1 || nameTwo.size() > 1 || nameThree.size() > 1 || nameFour.size() > 1;
+    }
+
+    protected static boolean doNamesContainSingleCharacterOnly(ArrayList<String> nameOne, ArrayList<String> nameTwo){
+        return nameOne.get(0).length() == 1 || nameTwo.get(0).length() == 1;
+    }
+
+    protected static AnswerType getHyphinatedNamesMatchResults(ArrayList<String> hNameOne, ArrayList<String> hNameTwo){
+        if (getNumOfCharsInName(hNameOne) > hyphenCharLimitYesVsMaybe && getNumOfCharsInName(hNameTwo) > hyphenCharLimitYesVsMaybe){
+            return AnswerType.yes;
+        }
+        return AnswerType.maybe;
+    }
+
+    protected static AnswerType getFirstLastSwapNameMatchResult(ArrayList<String> contactOneFirstName, ArrayList<String> contactOneLastName, ArrayList<String> contactTwoFirstName, ArrayList<String> contactTwoLastName){
+        // Check if both name sets are exact match
+        if (isNamesExactMatch(contactOneFirstName, contactTwoLastName) || isNamesExactMatch(contactTwoFirstName, contactOneLastName)){
+            return AnswerType.maybe;
+        }
+
+        // Check names with hyphen
+        if (doNamesContainMultipleParts(contactOneFirstName, contactTwoFirstName, contactOneLastName, contactTwoLastName) &&
+                (isNameMatchWithHyphens(contactOneFirstName, contactTwoLastName) && isNameMatchWithHyphens(contactTwoFirstName, contactOneLastName))
+                ){
+            return AnswerType.maybe;
+        }
+
+        // Check names with apostrophe
+        if (doNamesContainMultipleParts(contactOneFirstName, contactTwoFirstName, contactOneLastName, contactTwoLastName) &&
+                (combineNameParts(contactOneFirstName).equals(combineNameParts(contactTwoLastName)) && (combineNameParts(contactTwoFirstName).equals(combineNameParts(contactOneLastName))))
+                ){
+            return AnswerType.yes;
+        }
+
+        return AnswerType.no;
+    }
+
+    protected static boolean isNameMatchWithHyphens(ArrayList<String> nameOne, ArrayList<String> nameTwo){
+        if (nameTwo.size() > nameOne.size()){
```

Comment:
Title of method doesn't match what is happening in the method.

Reply:
Since the normalizeName() removed the actual hyphens and replaced them with spaces, and our testcases containing Hyphened names are only parts of the full name... This code treats the hyphenated testcases. That being said, if we can think of a better name, lets do it.

Reply:
How about "doNamePartsMatch"?

Reply:
Works for me :)